### PR TITLE
[Cleanup] Remove SetIconImage, GetIconImage, HasIcon and SetThumbnailImage

### PR DESCRIPTION
**Pull request by enen92**
## Description
Time to remove more annoying stuff. We've been asking addon developers since Jarvis to remove the following methods in favor of the `art` alternatives:

`xbmcgui.Listitem().SetIconImage(x)` -> `xbmcgui.Listitem().setArt({"icon": x})`
`xbmcgui.Listitem().SetThumbnailImage(y)` -> `xbmcgui.Listitem().setArt({"thumb": y})`
`xbmcgui.Listitem(iconImage=x, thumbnailImage=y)` -> `xbmcgui.Listitem().setArt()`

The PR also drops the `GetIconImage` and `HasIcon` in favor of `GetArt("icon")` and `HasArt("icon")`.

@AlwinEsch do you know if any of those methods is in use by binary addons?
@ksooo are they used in PVR?

## Motivation and Context
Cleanup deprecated python methods.

## How Has This Been Tested?
Compiled, runtime tested, check if behaviour stays the same

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [x] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [x] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [x] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Review by ronie — APPROVED** [review]

**Review by ksooo — COMMENTED** [review]
GUI API version bump missing? Every bin addon using the API must be recompiled after this change, otherwise it will crash if it uses the removed callbacks or callbacks which are located past the removed callbacks in the callback function table.

**Comment by enen92** [comment]
`xbmc.gui` bumped to 5.14.1

**Comment by ksooo** [comment]
Micro version change for breaking API change is IMO not okay. Should be at least a minor version number bump.

**Comment by ksooo** [comment]
Oh, and I'm not sure you touched the right file for the API version bump. I think this one is the correct file: https://github.com/xbmc/xbmc/blob/master/xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h

**Comment by ksooo** [comment]
=> https://github.com/xbmc/xbmc/blob/master/xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h#L47

**Comment by enen92** [comment]
Should `ADDON_GLOBAL_VERSION_GUI_MIN` also be changed?

**Comment by ksooo** [comment]
I think so, yes.

**Comment by enen92** [comment]
Done. I guess you're right, `xbmc.gui` version is for skins

**Comment by ksooo** [comment]
Yikes. Please rebase your branch to master branch, but do not merge master branch into your branch.

**Comment by enen92** [comment]
@ksooo I know merge commits aren't accepted, just give me time.

**Comment by ksooo** [comment]
git fetch upstream
git rebase upstream/master

File: xbmc/interfaces/legacy/ListItem.cpp [hunk 1]
```diff
@@ -49,7 +49,7 @@ namespace XBMCAddon
       if (!label2.empty())
         item->SetLabel2( label2 );
       if (!iconImage.empty())
-        item->SetIconImage( iconImage );
+        item->SetArt("icon", iconImage );
```
Comment (Rechi) [inline xbmc/interfaces/legacy/ListItem.cpp]: remove the space before `)`

File: xbmc/network/GUIDialogAccessPoints.cpp [hunk 2]
```diff
@@ -92,7 +92,7 @@ void CGUIDialogAccessPoints::OnInitWindow()
       else if (q <= 100) item->SetArt("thumb", "ap-signal5.png");
 
       if (m_aps[i].getEncryptionMode() != ENC_NONE)
-         item->SetIconImage("ap-lock.png");
+         item->SetArt("icon", "ap-lock.png");
```
Comment (Rechi) [inline xbmc/network/GUIDialogAccessPoints.cpp]: wrong indentation

File: xbmc/video/GUIViewStateVideo.cpp [hunk 3]
```diff
@@ -379,7 +379,7 @@ VECSOURCES& CGUIViewStateWindowVideoNav::GetSources()
     CMediaSource share;
     share.strName=item->GetLabel();
     share.strPath = item->GetPath();
-    share.m_strThumbnailImage= item->GetIconImage();
+    share.m_strThumbnailImage= item->GetArt("icon");
```
Comment (Rechi) [inline xbmc/video/GUIViewStateVideo.cpp]: space before `=` is missing

File: xbmc/interfaces/legacy/ListItem.cpp (hunk 1 above)
Comment (enen92) [reply xbmc/interfaces/legacy/ListItem.cpp]: Line is removed by a subsequent commit

File: xbmc/network/GUIDialogAccessPoints.cpp (hunk 2 above)
Comment (enen92) [reply xbmc/network/GUIDialogAccessPoints.cpp]: done

File: xbmc/video/GUIViewStateVideo.cpp (hunk 3 above)
Comment (enen92) [reply xbmc/video/GUIViewStateVideo.cpp]: done

File: xbmc/interfaces/legacy/ListItem.cpp (hunk 1 above)
Comment (Rechi) [reply xbmc/interfaces/legacy/ListItem.cpp]: The format should be fixed in the commit changing the line, even if it's removed in a later commit.

Comment (enen92) [reply xbmc/interfaces/legacy/ListItem.cpp]: Truly important. Done.

Comment (ksooo) [reply xbmc/interfaces/legacy/ListItem.cpp]: Long live the bureaucracy. ;-)

**Review by AlwinEsch — APPROVED** [review]
From my side OK. The `SetArt` alone is also good and this prevent some doubles.

By a search over addons has not found the use of them.

**Comment by enen92** [comment]
Due to objections to the actual state of this PR (API breakage) I'll superseed this with another PR where the callbacks and api methods are kept but resulting in NOP for Matrix. I hope that at some time during N* release cycle we'll be able to simple remove the affected methods.

**Pull request closed without merging**